www/runtest.php

```diff
@@ -538,7 +538,19 @@ function DealWithMagicQuotes(&$arr) {
                 $test['customMetrics'][$metric] = base64_encode($code);
               }
             }
-            
+
+            if (array_key_exists('heroElements', $_REQUEST)) {

```

Comment:
Is there a mode where the hero elements can be auto-detected (largest atf image, etc) or is the agent always going to do that independently of heroElements being specified?  I kind of expected it to find the default elements but override it if specific elements are passed in.

Also, a global setting to turn it on/off across the install and a per-test option (expecting it to add some processing overhead so not everyone may want to turn it on).  My vote would be for enabled by default with an option to turn it off.

Reply:
There are some default elements in [hero_elements.js](https://github.com/wildlyinaccurate/wptagent/blob/visualmetrics-hero-element-times/internal/js/hero_elements.js#L24-L40) which lives on the agents. Specifically the defaults are largest H1 (with H2 as a fallback), largest IMG, largest element with a background image, and any element with the `elementtiming` attribute.

This `heroElements` option is so that you can specify selectors for hero elements at test time, in the case that you can't (or don't want to) add the `elementtiming` attribute.